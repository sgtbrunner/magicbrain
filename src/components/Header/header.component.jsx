import React from 'react';
import PropTypes from 'prop-types';

import './header.styles.css';

const Header = ({ onRouteChange, signedIn }) =>
  signedIn ? (
    <nav className="header">
      <button
        className="f4 link dim black underline pa4 pointer noselect"
        onClick={() => onRouteChange('signin')}
        type="button"
      >
        Sign Out
      </button>
    </nav>
  ) : (
    <nav className="header">
      <button
        className="f4 link dim black underline pa3 pointer noselect"
        onClick={() => onRouteChange('signin')}
        type="button"
      >
        Sign In
      </button>
      <button
        className="f4 link dim black underline pa3 pointer noselect"
        onClick={() => onRouteChange('register')}
        type="button"
      >
        Register
      </button>
    </nav>
  );

Header.propTypes = {
  onRouteChange: PropTypes.func.isRequired,
  signedIn: PropTypes.bool.isRequired,
};

export default Header;
