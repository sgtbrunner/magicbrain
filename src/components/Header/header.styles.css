.header {
  display: flex;
  justify-content: flex-end;
  padding-top: 20px;
}

.header button {
  border: none;
  background-color: transparent;
}
