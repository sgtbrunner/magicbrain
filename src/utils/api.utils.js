const { REACT_APP_API_ENDPOINT } = process.env;

const fetchServer = ({
  path,
  method = 'POST',
  headers = { 'Content-Type': 'application/json' },
  body,
}) =>
  fetch(`${REACT_APP_API_ENDPOINT}${path}`, {
    method,
    headers,
    body: JSON.stringify({ ...body }),
  });

export default {
  signInUser: ({ email, password }) =>
    fetchServer({ path: '/signin', body: { email, password } }).then((response) => response.json()),

  registerUser: ({ name, email, password }) =>
    fetchServer({ path: '/register', body: { name, email, password } }).then((response) =>
      response.json()
    ),

  getImageBoundary: ({ input }) =>
    fetchServer({ path: '/imageurl', body: { input } }).then((response) => response.json()),

  updateImageCount: ({ id }) =>
    fetchServer({ path: '/image', method: 'PUT', body: { id } }).then((response) =>
      response.json()
    ),
};
