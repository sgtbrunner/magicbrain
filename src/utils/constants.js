export const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export const { REACT_APP_API_ENDPOINT } = process.env;
