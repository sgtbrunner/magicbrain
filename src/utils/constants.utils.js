const PARTICLE_OPTIONS = {
  particles: {
    number: {
      value: 50,
      density: {
        enable: true,
        value_area: 500,
      },
    },
  },
};

export default PARTICLE_OPTIONS;
