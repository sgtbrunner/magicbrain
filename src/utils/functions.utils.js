export const getUserFromLocalStorage = () => localStorage.getItem('user');
