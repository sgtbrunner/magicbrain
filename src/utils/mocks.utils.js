export const USER = {
  id: 3,
  name: 'test',
  email: 'test@test.com',
  entries: 7,
};

export const OUTPUTS = {
  outputs: [
    {
      data: {
        regions: [
          {
            region_info: {
              bounding_box: {
                top_row: 0.5,
                left_col: 0.5,
                bottom_row: 0.5,
                right_col: 0.5,
              },
            },
          },
        ],
      },
    },
  ],
};

export const CREDENTIAL_ERROR = {
  error: 'You have entered an invalid username and/or password',
};

export const SERVER_ERROR = {
  error: 'Something went wrong. Please try again later.',
};
