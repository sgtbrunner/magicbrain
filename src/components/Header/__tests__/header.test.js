import { screen } from '@testing-library/react';

import { mountComponent } from '../../../utils/test.utils';
import Header from '..';

const BUTTON = 'button';
const clearUser = jest.fn();

const mountHeaderComponent = ({ isUserSignedIn }) =>
  mountComponent({ component: Header, props: { isUserSignedIn, clearUser } });

describe('Header component', () => {
  it('should match snapshot when user IS NOT signed id', () => {
    const { container } = mountHeaderComponent({ isUserSignedIn: false });

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when user IS signed id', () => {
    const { container } = mountHeaderComponent({ isUserSignedIn: true });

    expect(container).toMatchSnapshot();
  });

  it('should render two (2) buttons when user IS NOT signed in', () => {
    mountHeaderComponent({ isUserSignedIn: false });

    const buttons = screen.getAllByRole(BUTTON);
    expect(buttons).toHaveLength(2);
  });

  it(`should render 'Sign In' and 'Register' buttons when user IS NOT signed in`, () => {
    mountHeaderComponent({ isUserSignedIn: false });

    const signInButton = screen.getByRole(BUTTON, { name: 'Sign In' });
    expect(signInButton).toBeInTheDocument();

    const registerButton = screen.getByRole(BUTTON, { name: 'Register' });
    expect(registerButton).toBeInTheDocument();
  });

  it('should render one (1) button when user IS signed in', () => {
    mountHeaderComponent({ isUserSignedIn: true });

    const buttons = screen.getAllByRole(BUTTON);
    expect(buttons.length).toBe(1);
  });

  it(`should render 'Sign out' button when user IS signed in`, () => {
    mountHeaderComponent({ isUserSignedIn: true });

    const signOutButton = screen.getByRole(BUTTON, { name: 'Sign Out' });
    expect(signOutButton).toBeInTheDocument();
  });
});
