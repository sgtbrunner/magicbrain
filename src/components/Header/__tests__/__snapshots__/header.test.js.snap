// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot when user IS NOT signed id 1`] = `
<div>
  <nav
    class="flex justify-end"
  >
    <div>
      <button
        class="f4 link black pv2 ma3 b--none bg-transparent"
        type="button"
      >
        Sign In
      </button>
      <button
        class="f4 link black pv2 ma3 b--none bg-transparent"
        type="button"
      >
        Register
      </button>
    </div>
  </nav>
</div>
`;

exports[`Header component should match snapshot when user IS signed id 1`] = `
<div>
  <nav
    class="flex justify-end"
  >
    <button
      class="f4 link black pv2 ma3 b--none bg-transparent"
      type="button"
    >
      Sign Out
    </button>
  </nav>
</div>
`;
